custom_components/unifiprotect/binary_sensor.py

```diff
@@ -129,17 +180,18 @@ def __init__(
         protect_data,
         server_info,
         device_id,
-        sensor_type,
-        sensor,
+        description: UnifiProtectBinaryEntityDescription,
     ):
         """Initialize the Binary Sensor."""
-        super().__init__(upv_object, protect_data, server_info, device_id, sensor_type)
-        self._name = f"{sensor_type.capitalize()} {self._device_data['name']}"
+        super().__init__(upv_object, protect_data, server_info, device_id, description)
+        self.entity_description = description
+        self._name = (
+            f"{self.entity_description.name.capitalize()} {self._device_data['name']}"
+        )
         self._attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
-        if self._device_data["type"] == DEVICE_TYPE_SENSOR:
-            self._device_class = sensor_type
-        else:
-            self._device_class = PROTECT_TO_HASS_DEVICE_CLASS.get(sensor_type)
+        self._device_class = self.entity_description.device_class
+        self._icon = self.entity_description.icon
```

Comment:
You should be able to drop `def icon` as well

Reply:
So both device_class and icon will be picked up automatically when I the xxxEntityDescription?

Reply:
Smart. I will make more edits tomorrow morning. 
Thanks